src/com/gfredericks/test/chuck/regexes.clj

```diff
@@ -157,16 +157,16 @@
                             {:type :group
                              :elements [alternation]})
                          ([group-flags alternation]
-                          (cond-> {:type :group
-                                   :elements [alternation]
-                                   :flag group-flags}
-                            ;; If the flags are just an empty
-                            ;; NonCapturingMatchFlags, we can safely
-                            ;; ignore it; this should correspond to
-                            ;; something like #"foo(?:bar)*"
-                            (not= group-flags
-                                  [:GroupFlags [:NonCapturingMatchFlags [:MatchFlagsExpr]]])
-                            (assoc :unsupported #{:flags}))))
+                          (let [parsed {:type :group
+                                        :elements [alternation]
+                                        :flag group-flags}
+                                [flag-header [flag-type flag-details]] group-flags]
```

Comment:
For your reviewing comfort, when you have a NonCapturingMatch, group-flags looks like:

```clj
[:GroupFlags [:NonCapturingMatchFlags [:MatchFlagsExpr]]]
```

When you have a NamedCapturingGroup, it looks like:

```clj
[:GroupFlags [:NamedCapturingGroup [:GroupName "x" ] ] ]
```

Reply:
I think you changed the logic here so that `#"foo(?i:bar)" is accepted, when it used to be `:unsupported`. I'll edit it to use the full data structure check from the old version.